src/otserv.cpp

```diff
@@ -37,96 +33,67 @@
-Monsters g_monsters;
-Vocations g_vocations;
-extern Scripts* g_scripts;
-RSA g_RSA;
+Game g_game;			// Holds and mutates game state (map, players, monsters, NPCs, etc)
+ConfigManager g_config; // Holds configuration from '/config.lua'
+Monsters g_monsters;	// Holds monster definitions from /data/monster (XMLs)
+Vocations g_vocations;	// Holds vocations definitions from '/data/XML/vocations.xml'
+RSA g_RSA;				// Holds private key from '/key.pem' and decrypts data
 
+// Prevents 'main()' from continuing execution without 'mainLoader()'
 std::mutex g_loaderLock;
 std::condition_variable g_loaderSignal;
 std::unique_lock<std::mutex> g_loaderUniqueLock(g_loaderLock);
 
-void startupErrorMessage(const std::string& errorStr)
-{
-	std::cout << "> ERROR: " << errorStr << std::endl;
-	g_loaderSignal.notify_all();
-}
-
-void mainLoader(int argc, char* argv[], ServiceManager* services);
+/* Entry point 'main()' at the end of the file. */
 
 [[noreturn]] void badAllocationHandler()
 {
-	// Use functions that only use stack allocation
 	puts("Allocation failed, server out of memory.\nDecrease the size of your map or compile in 64 bits mode.\n");
 	getchar();
 	exit(-1);
 }
 
-int main(int argc, char* argv[])
+void startupErrorMessage(const std::string& errorStr)
 {
-	// Setup bad allocation handler
-	std::set_new_handler(badAllocationHandler);
-
-	ServiceManager serviceManager;
-
-	g_dispatcher.start();
-	g_scheduler.start();
-
-	g_dispatcher.addTask(createTask(std::bind(mainLoader, argc, argv, &serviceManager)));
-
-	g_loaderSignal.wait(g_loaderUniqueLock);
-
-	if (serviceManager.is_running()) {
-		std::cout << ">> " << g_config.getString(ConfigManager::SERVER_NAME) << " Server Online!" << std::endl << std::endl;
-		serviceManager.run();
-	} else {
-		std::cout << ">> No services running. The server is NOT online." << std::endl;
-		g_scheduler.shutdown();
-		g_databaseTasks.shutdown();
-		g_dispatcher.shutdown();
-	}
-
-	g_scheduler.join();
-	g_databaseTasks.join();
-	g_dispatcher.join();
-	return 0;
+	std::cout << "> ERROR: " << errorStr << std::endl;
+	g_loaderSignal.notify_all();
 }
 
-void mainLoader(int, char*[], ServiceManager* services)
-{
-	//dispatcher thread
+void mainLoader(int argc, char* argv[], ServiceManager* services);
+
+void mainLoader(int, char* [], ServiceManager* services) {
 	g_game.setGameState(GAME_STATE_STARTUP);
 
 	srand(static_cast<unsigned int>(OTSYS_TIME()));
-#ifdef _WIN32
+#	ifdef _WIN32
 	SetConsoleTitle(STATUS_SERVER_NAME);
-#endif
+#	endif
 	std::cout << STATUS_SERVER_NAME << " - Version " << STATUS_SERVER_VERSION << std::endl;
 	std::cout << "Compiled with " << BOOST_COMPILER << std::endl;
 	std::cout << "Compiled on " << __DATE__ << ' ' << __TIME__ << " for platform ";
 
-#if defined(__amd64__) || defined(_M_X64)
+#	if defined(__amd64__) || defined(_M_X64)
 	std::cout << "x64" << std::endl;
-#elif defined(__i386__) || defined(_M_IX86) || defined(_X86_)
+#	elif defined(__i386__) || defined(_M_IX86) || defined(_X86_)
```

Comment:
Whats the deal with a random tab between the # and conditional declaration?

Reply:
It's not random. The tab indents the directives into the same block as the inserted code would be.
We could also just indent the whole line, but then we wouldn't easily see where there's preprocessing.

Reply:
The # symbol is not indented into the correct scope level, which breaks the scope collapse functionality in text editors such as Sublime Text.
![image](https://user-images.githubusercontent.com/1552144/67578680-42607d00-f743-11e9-86a3-99cabffebd3b.png)